samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -109,3 +117,60 @@ def _get_configuration_address(address: str) -> str:
         The address clean of indices
     """
     return re.sub(r"\[[^\[\]]*\]", "", address)
+
+
+def _resolve_module_output(module, output_name):
+    pass
+
+
+def _resolve_module_variable(module: TFModule, variable_name: str) -> List[Union[ConstantValue, ResolvedReference]]:
+    # return a list of the values that resolve the passed variable
+    # name in the input module.
+    results: List[Union[ConstantValue, ResolvedReference]] = []
+
+    LOG.debug("Resolving module variable for module (%s) and variable (%s)", module.module_name, variable_name)
+
+    var_value = module.variables.get(variable_name)
+
+    if not var_value:
+        raise InvalidResourceLinkingException(
+            message=f"The variable {variable_name} could not be found in module {module.module_name}."
+        )
+
+    # check the possible constant value for this variable
+    if isinstance(var_value, ConstantValue) and var_value is not None:
+        LOG.debug("Found a constant value (%s) in module (%s)", var_value.value, module.module_name)
+        results.append(ConstantValue(var_value.value))
+
+    # check the possible references value for this variable
+    if isinstance(var_value, References) and var_value is not None:
+        LOG.debug("Found references (%s) in module (%s)", var_value.value, module.module_name)
+        cleaned_references = _clean_references_list(var_value.value)
+        for reference in cleaned_references:
+            LOG.debug("Resolving reference: %s", reference)
+            # refer to a variable passed to this module from its parent module
+            if reference.startswith("var."):
+                config_var_name = _get_configuration_address(reference[4:])
```

Comment:
`4:` should we look at len("var.") instead?

Reply:
I like this for readability. Updated.